examples/platforms/simulation/platform-simulation.h

```diff
@@ -67,6 +67,8 @@ enum
     OT_SIM_EVENT_UART_WRITE         = 2,
     OT_SIM_EVENT_RADIO_SPINEL_WRITE = 3,
     OT_SIM_EVENT_OTNS_STATUS_PUSH   = 5,
+    OT_SIM_EVENT_RADIO_COMM         = 6,
```

Comment:
In my design, there are two separate events: OT_SIM_EVENT_RADIO_RX (received by OT node) and OT_SIM_EVENT_RADIO_TX (sent by OT node) which is useful if the two events are accompanied by different datastructures in the payload.  Since for a transmit partly different data is used (e.g. Tx Power) than for a receive (e.g. RSSI).  The event serialization / deserialization code in OTNS (see https://github.com/EskoDijk/ot-ns/blob/sim-radiomodel/types/event.go ) in the long run can be kept simpler if 2 different event types are used.

Reply:
I liked your design of using a data struct for the messages. However, do we really need two events to signal communication, when in fact, any income can be treated as a Rx, and every output can be treated as a Tx?
I see no difference to have one or two signals, when they are never mixed anyway.

Reply:
> any income can be treated as a Rx, and every output can be treated as a Tx

For the OT code here I agree, but for the Go OTNS side this will be needed. For example I have a PR for an event-dump command, and for printing events it uses the type to decide how to display it - relevant code part is here https://github.com/EskoDijk/ot-ns/blob/pr-dump-events/types/event.go#L168 

So the code in the Event class needs to decide, purely based on type, how to print it and what the relevant data-of-the-event is. You could add this as an extra boolean but then the most logical way to do this is encode it in the type. I.e. different event types have different payloads, a single event type has a single defined payload format; makes most sense.

Reply:
I still see that even from the OTNS, it will never send and event with a TX signal, unless you were using that as a TxDone signal (which you are not). I understand that you would need a boolean, that is what I would have done, but honestly, this is just a style decision. Both work well and the same. I guess @simonlingoogle could give a final word on what they see it closer to their designs. I am ok having either way.